models/surveys.go

```diff
@@ -486,13 +493,42 @@ func (api *API) Info(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-// AllSurveys returns summaries of all known surveys. The surveys are returned in ascending short name order.
+//AllSurveys handler returns a list of all surveys
 func (api *API) AllSurveys(w http.ResponseWriter, r *http.Request) {
-	rows, err := api.AllSurveysStmt.Query()
+	api.GetSurveys(w, r, "")
+}
 
-	if err != nil {
-		http.Error(w, "AllSurveys query failed", http.StatusInternalServerError)
-		return
+//SurveysByType handler returns a list of all surveys of a particular type e.g Business or Social
+func (api *API) SurveysByType(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	surveyType := vars["surveyType"]
+
+	// Mux package does not pass survey type during tests
+	if surveyType == "" {
+		surveyType = "Business"
+	}
+	api.GetSurveys(w, r, surveyType )
+}
+
+// GetSurveys returns summaries of all known surveys. The surveys are returned in ascending short name order.
+// optional: surveyType if supplied then the list only returns surveys which match that surveyType
+// valid surveyTypes are 'Business', 'Social' or 'Census'
+func (api *API) GetSurveys(w http.ResponseWriter, r *http.Request, surveyType string) {
+	var rows *sql.Rows
+	var err error
+
+	if len(surveyType) == 0 {
+		rows, err = api.AllSurveysStmt.Query()
+		if err != nil {
+			http.Error(w, "AllSurveys query failed", http.StatusInternalServerError)
+			return
+		}
+	} else{
+		rows, err = api.GetSurveysBySurveyTypeStmt.Query(surveyType)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("AllSurveys query for surveyType %s failed", surveyType), http.StatusInternalServerError)
```

Comment:
Don't need to go into internal details when returning an error to the client. Would be more useful to return a more end-user-friendly error like `failed to retrieve surveys` and log the more explicit reasons instead.

In either case, this is saying it's an `AllSurveys` call when it's actually filtered.